docs/core/diagnostics/metrics-collection.md

```diff
@@ -0,0 +1,404 @@
+---
+title: Collect metrics - .NET
+description: Tutorial to collect metrics in .NET applications
+ms.topic: tutorial
+ms.date: 10/27you/2021
+---
+
+# Collect metrics
+
+**This article applies to: ✔️** .NET Core 3.1 and later versions **✔️** .NET Framework 4.5 and later versions
+
+Instrumented code can record numeric measurements, but the measurements usually need to be aggregated, transmitted,
+and stored to create useful metrics for monitoring. This process of aggregating, transmitting and storing the data is
+called collection. In this tutorial, we will show several examples on how to collect metrics:
+  - Populating metrics in Grafana with OpenTelemetry and Prometheus
+  - Viewing metrics in real-time with the dotnet-counters command-line tool
+  - Creating a custom collection tool using the underlying .NET <xref:System.Diagnostics.Metrics.MeterListener> API
+
+For more information on custom metric instrumentation see [here](./understanding-different-metric-apis.md) for an overview
+of instrumentation options.
+
+### Create an example application
+
+**Prerequisites**: [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet) or a later version
+
+Before metrics can be collected, we need to produce some measurements. For simplicity we will create a small app that has
+some trivial metric instrumentation. The .NET runtime also has [a variety of metrics of built-in](available-counters.md). See 
+[the instrumentation tutorial](metrics-instrumentation.md) for more details about creating new metrics using the
+<xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> API shown here.
+
+```dotnetcli
+dotnet new console
+dotnet add package System.Diagnostics.DiagnosticSource
+```
+
+Replace the code of Program.cs with:
+```C#
+using System;
+using System.Diagnostics.Metrics;
+using System.Threading;
+
+class Program
+{
+    static Meter s_meter = new Meter("HatCo.HatStore", "1.0.0");
+    static Counter<int> s_hatsSold = s_meter.CreateCounter<int>("hats-sold"); 
+
+    static void Main(string[] args)
+    {
+        Console.WriteLine("Press any key to exit");
+        while(!Console.KeyAvailable)
+        {
+            // Pretend our store has a transaction each second that sells 4 hats
+            Thread.Sleep(1000);
+            s_hatsSold.Add(4);
+        }
+    }
+}
+```
+
+### Viewing metrics with dotnet-counters
+
+[dotnet-counters](dotnet-counters.md) is a simple command-line tool that can view live metrics for any .NET Core application on
+demand. It doesn't require any advance setup which can make it very useful for ad-hoc investigations or to verify that metric
+instrumentation is working correctly. It works with both <xref:System.Diagnostics.Metrics?displayProperty=nameWithType> based
+APIs as well as [EventCounters](event-counters.md).
+
+If the [dotnet-counters](./dotnet-counters.md) tool is not already installed use the SDK to install it:
+
+```dotnetcli
+> dotnet tool update -g dotnet-counters
```

Comment:
>dotnet tool update -g dotnet-counters [](http://example.com/codeflow?start=2&length=37)

just checking, if anyone using .NET Core 3.1 and run this command, still can get the latest version of the tool. is that right?

Reply:
Yep, I did all the testing with only the 3.1 SDK + runtime installed to confirm no later dependencies crept in:

```
> dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.414
 Commit:    88a3b0c5ca

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.414\

Host (useful for support):
  Version: 3.1.20
  Commit:  2833dab8c3

.NET Core SDKs installed:
  3.1.414 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

> dotnet tool uninstall dotnet-counters -g
Tool 'dotnet-counters' (version '5.0.251802') was successfully uninstalled.

> dotnet tool update -g dotnet-counters
You can invoke the tool using the following command: dotnet-counters
Tool 'dotnet-counters' (version '5.0.251802') was successfully installed.
```